backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/model/validator/AsyncBatchNotificationHelper.java

```diff
@@ -0,0 +1,127 @@
+import de.metas.async.api.IWorkPackageQueue;
+import de.metas.async.model.I_C_Async_Batch;
+import de.metas.async.processor.IWorkPackageQueueFactory;
+import de.metas.organization.ClientAndOrgId;
+import de.metas.printing.async.spi.impl.InvoicePDFConcatenateWorkpackageProcessor;
+import de.metas.printing.model.I_C_Printing_Queue;
+import de.metas.util.Services;
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.dao.IQueryBuilder;
+import org.adempiere.ad.dao.IQueryFilter;
+import org.adempiere.ad.dao.impl.TypedSqlQueryFilter;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.service.ISysConfigBL;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static de.metas.async.Async_Constants.C_Async_Batch_InternalName_AutomaticallyInvoicePdfPrinting;
+
+@UtilityClass
+public class AsyncBatchNotificationHelper
+{
+	private final IQueryBL iQueryBL = Services.get(IQueryBL.class);
+	private final IWorkPackageQueueFactory workPackageQueueFactory = Services.get(IWorkPackageQueueFactory.class);
+	private final IAsyncBatchListeners asyncBatchListeners = Services.get(IAsyncBatchListeners.class);
+	private final IAsyncBatchBL asyncBatchBL = Services.get(IAsyncBatchBL.class);
+	private final ISysConfigBL sysConfigBL = Services.get(ISysConfigBL.class);
+
+	protected void notify(final I_C_Async_Batch asyncBatch)
+	{
+		asyncBatchListeners.notify(asyncBatch);
+	}
+
+	protected void runPrintingProcess(@NonNull final I_C_Async_Batch asyncBatch)
+	{
+		List<IQueryBuilder<I_C_Printing_Queue>> queries = fetchPrintingQueues(asyncBatch);
+		for (final IQueryBuilder<I_C_Printing_Queue> pqs : queries)
+		{
+			enqueuePrintQueues(pqs, asyncBatch);
+		}
+	}
+
+	private void enqueuePrintQueues(@NonNull final IQueryBuilder<I_C_Printing_Queue> pqs, @NonNull final I_C_Async_Batch parentAsyncBatch)
+	{
+		final List<I_C_Printing_Queue> printing_queues = pqs.create().list();
+		if (printing_queues.isEmpty())
+		{
+			return;
+		}
+
+		final AsyncBatchId asyncBatchId = createAsyncBatch();
+		final Properties ctx = InterfaceWrapperHelper.getCtx(parentAsyncBatch);
+
+		final IWorkPackageQueue queue = workPackageQueueFactory.getQueueForEnqueuing(ctx, InvoicePDFConcatenateWorkpackageProcessor.class);
+		queue.newBlock()
+				.setContext(ctx)
+				.newWorkpackage()
+				.setC_Async_Batch(asyncBatchBL.getAsyncBatchById(asyncBatchId))
+				.addElements(printing_queues)
+				.build();
+	}
+
+	private AsyncBatchId createAsyncBatch()
+	{
+		return asyncBatchBL.newAsyncBatch(C_Async_Batch_InternalName_AutomaticallyInvoicePdfPrinting);
+	}
+
+	private List<IQueryBuilder<I_C_Printing_Queue>> fetchPrintingQueues(@NonNull final I_C_Async_Batch asyncBatch)
+	{
+		final Map<String, String> valuesForPrefix = sysConfigBL.getValuesForPrefix("UserQuery_", ClientAndOrgId.SYSTEM);
```

Comment:
- unless there is a particular reason not to, please create this sysconfig on org-level, so different orgs may have different user queries
- looks like there aren't `IUserQuery`s but whereclauses. 
  => pleaserename to something with a prefix of `de.metas.printing.` that indicates when the where clauses are used.
- also, please create one inactive sysconfig in metasfresh so future users know where to start. make sure that this sysconfig contains a helpful description